java/code/src/com/suse/manager/webui/websocket/RemoteMinionCommands.java

```diff
@@ -265,13 +269,23 @@ else if (previewedMinions.isEmpty()) {
 
                 res.forEach((minionId, future) -> future.whenComplete((cmdResult, err) -> {
                     if (cmdResult != null) {
-                        sendMessage(session,
-                                cmdResult.fold(
-                                        error -> new ActionErrorEventDto(minionId,
-                                                "ERR_CMD_SALT_ERROR",
-                                                parseSaltError(error)),
-                                        result -> new MinionCommandResultEventDto(minionId,
-                                                result)));
+                        AbstractSaltEventDto event = cmdResult.fold(
+                                error -> {
+                                    String errMsg = parseSaltError(error);
+                                    LOG.info("Received Salt error for minion " + minionId + ": " + errMsg);
+                                    return new ActionErrorEventDto(minionId,
+                                        "ERR_CMD_SALT_ERROR", errMsg);
+                                },
+                                result -> {
+                                    LOG.info("User '" + webSession.getUser().getLogin() +
+                                            "' received command result from minion " + minionId +
+                                            ". Output is logged at DEBUG level.");
+                                    if (LOG.isDebugEnabled()) {
+                                        LOG.debug("Minion " + minionId + " returned:\n" + result);
+                                    }
```

Comment:
I wonder if it would be also easy to add this to the event history of the minion which executed this command.

```
            ServerHistoryEvent event = new ServerHistoryEvent();
            event.setCreated(new Date());
            event.setServer(MinionServerFactory.findByMinionId(minionId));
            event.setSummary("Salt Remote Command");
            event.setDetails("Minion " + minionId + " returned:\n" + result);
            server.getHistory().add(event);
```

or something like this.

Reply:
@mcalmer I've added an event to the system history. Please take a look at the screenshots.